### PR TITLE
Remove map.h from game.h

### PR DESCRIPTION
**Pull request by Coolthulhu**
Unraveled one of the biggest spaghettification of header inclusion in the game.
This required changing `game::m` to a reference and creating a pointer to back it up. This is a little bit hacky, but worth not having almost the entirety of code recursively include ~80% of headers. `map.h` contains a lot of inclusions of big headers, like `player.h` and `item.h`.

EDIT: Removed `npc.h` from `map.h`, making it lose all the creature-related code inclusions.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -92,8 +93,10 @@ bool is_valid_in_w_terrain(int x, int y)
 
 // This is the main game set-up process.
 game::game() :
+    map_ptr( new map() ),
     new_game(false),
     uquit(QUIT_NO),
+    m( *map_ptr.get() ),
```
Comment (BevapDin) [inline src/game.cpp]: `.get()` is not really required, `unique_ptr` has a `operator*` which returns the desired reference.

cata_tiles.cpp also needs the map.h include.

Comment (Coolthulhu) [inline src/game.cpp]: cata_tiles.cpp seems to already have it explicitly.

**Comment by BevapDin** [comment]
Compiling sdltiles.cpp (with TILES=1) fails with
```
In file included from src/sdltiles.cpp:15:0:
src/cata_tiles.h:218:57: Fehler: »lit_level« bezeichnet keinen Typ
         void draw_single_tile( const tripoint &p, const lit_level ll,
                                                         ^
src/cata_tiles.h:220:56: Fehler: »visibility_type« bezeichnet keinen Typ
         bool apply_vision_effects( int x, int y, const visibility_type visibility);
                                                        ^
```
You might need to forward declare that enum. cata_tiles.cpp fails for the same reason.

It also fails in newcharacter.cpp because of strcmp.

**Comment by Coolthulhu** [comment]
Oops, forgot to test the tiles build.
Added the entire map.h to cata_tiles.h for now. cata_tiles.h is only included in 4 .cpp files.

**Comment by Coolthulhu** [comment]
OK, got it to compile (and run) on tiles. At least with `g++ 4.9.1`.

**Comment by BevapDin** [comment]
The bot approves, mxe cross compilation works, and it does look correct anyway.

**Pull request merged**